pkg/policy/watcher/watcher.go

```diff
@@ -42,20 +42,38 @@ func (m *MetaWatcher) Run() {
 			m.logger.Debug().Msg("meta watcher last index has not changed")
 			continue
 		}
+		m.logger.Debug().
+			Uint64("old", q.WaitIndex).
+			Uint64("new", meta.LastIndex).
+			Msg("meta watcher last index has changed")
 
-		m.logger.Debug().Msg("meta watcher last index has changed")
+		// Update our last tracked index on this run to match the meta returned from the API.
```

Comment:
Looks like we are  updating the `maxfound` for every outer for loop iteration, and also in the inner for loop going over jobs. If the execution goes into the inner for-loop, the `maxfound` may actually decrease in value (also it looks like the `jobs` list is sorted based on the job `ID` instead of job `ModifyIndex`). This will probably result in `q.WaitIndex` to be lower than the one in the meta returned from the API call. That may create some cruft by making redundant api calls that won't block.

 It appears that `q.WaitIndex` and `m.lastChangeIndex` are keeping track of two separate things, so assigning them the sam `maxFound` at the end of the outer for might not be useful. Can we just use the `maxfound` to keep track of the last maximum index that results in loading the job policy params from the meta stanza for a job, and then assign that to `lastChangeIndex` while directly assigning `q.WaitIndex` -> `meta.LastIndex` at the end of the outer for loop.